### ISSUE TITLE
Bug: Error while loading rule 'logical-assignment-operators

### ISSUE THREAD

**Issue by electriquo** [eslint/markdown#482]
### Environment

Node version: v22.17.1
npm version: v11.4.2
Local ESLint version: v9.31.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.5.0

### Which language are you using?

gfm

### What did you do?

<details>
<summary>Configuration</summary>

```
// eslint.config.js
import { defineConfig } from "eslint/config";
import markdown from "@eslint/markdown";
import js from "@eslint/js";

export default defineConfig([
  js.configs.all,
  markdown.configs.recommended
]);
```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

Plugin should work.

### What actually happened?

```shell
Oops! Something went wrong! :(

ESLint: 9.31.0

TypeError: Error while loading rule 'logical-assignment-operators': sourceCode.getScope is not a function
Occurred while linting /Users/foo/prj/README.md
    at Object.create (/Users/foo/prj/node_modules/eslint/lib/rules/logical-assignment-operators.js:290:31)
    at createRuleListeners (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:1134:15)
    at /Users/foo/prj/node_modules/eslint/lib/linter/linter.js:1285:7
    at Array.forEach (<anonymous>)
    at runRules (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:1201:31)
    at #flatVerifyWithoutProcessors (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:2252:22)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:2343:43)
    at Linter._verifyWithFlatConfigArray (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:2446:15)
    at Linter.verify (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:1808:10)
    at Linter.verifyAndFix (/Users/foo/prj/node_modules/eslint/lib/linter/linter.js:2711:20)
```

### Link to Minimal Reproducible Example

https://github.com/electriquo/eslint-markdown-err

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
Because you are now linting Markdown files in addition to JavaScript files.

**Comment by nzakas** [comment]
Use `--inspect-config`
